[ONNX] Update slice process shape to support rank only inference (#65782)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#66149 [ONNX] Update slice process shape to support rank only inference (#65782)**
* #66147 [ONNX] Symbolic: dynamic input for OneHot, bool for Einsum (#65940)
* #66146 [ONNX] Modify softplus symbolic to support beta!=1 (#65001)
* #66145 [ONNX] Deprecate fold_if pass (#65697)
* #66144 [ONNX] Add diagonal symbolic (#64454)
* #66143 ONNX: Delete or document skipped ORT tests (#64470)
* #66142 [ONNX] Add warning for inplace updates on tensor.shape in tracing mode (#63170)
* #66141 [ONNX] Update sum symbolic to handle dtypes (#64289)
* #66140 [ONNX] Export nn.Module call as ONNX local function (#63589)

Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D31423840](https://our.internmc.facebook.com/intern/diff/D31423840)

BowenBao committed:
[ONNX] Update slice process shape to support rank only inference (#65782)

Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update slice process shape to support rank only inference (#65782)"


Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao commented:
@malfet please help import. Let me know what's the right handle to tag for notifying.

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31423840).

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31423840).

BowenBao committed:
Update on "[ONNX] Update slice process shape to support rank only inference (#65782)"


Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D31423840](https://our.internmc.facebook.com/intern/diff/D31423840)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update slice process shape to support rank only inference (#65782)"


Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D31423840](https://our.internmc.facebook.com/intern/diff/D31423840)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update slice process shape to support rank only inference (#65782)"


Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D31423840](https://our.internmc.facebook.com/intern/diff/D31423840)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update slice process shape to support rank only inference (#65782)"


Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D31423840](https://our.internmc.facebook.com/intern/diff/D31423840)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update slice process shape to support rank only inference (#65782)"


Updated logic will be able to infer rank of slice output, when only rank is known for slice input. Enables cases where `ConstantValueMap::HasRank(input)` is `True`, while `ConstantValueMap::HasShape(input)` is `False`.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D31423840](https://our.internmc.facebook.com/intern/diff/D31423840)

[ghstack-poisoned]
